rcl_yaml_param_parser/include/rcl_yaml_param_parser/types.h

```diff
@@ -74,12 +74,39 @@ typedef struct rcl_node_params_s
   size_t num_params;  ///< Number of parameters in the node
 } rcl_node_params_t;
 
+/// \typedef rcl_param_descriptor_t
+/// \brief param_descriptor_t stores the descriptor of a parameter
+typedef struct rcl_param_descriptor_s
+{
+  char * name;
+  bool * read_only;
+  uint8_t * type;
+  char * description;
+  char * additional_constraints;
+  double * min_value_double;
+  double * max_value_double;
+  double * step_double;
+  int64_t * min_value_int;
+  int64_t * max_value_int;
+  int64_t * step_int;
+} rcl_param_descriptor_t;
+
+/// \typedef rcl_node_params_descriptors_t
+/// \brief node_params_descriptors_t stores all the parameter descriptors of a single node
+typedef struct rcl_node_params_descriptors_s
+{
+  char ** parameter_names;  ///< Array of parameter names (keys)
```

Comment:
@bpwilcox I don't follow why there's an array of parameter names here and a name on each descriptor.

Reply:
Parallelism? I think I had intended to parallel the structure of `rcl_node_params` and `rcl_node_params_descriptors`. It made the code more convenient to write, I believe, though I suppose we could remove one or the other.

Reply:
Hmm I'm not entirely convinced it's worth the duplication. You can get away with just an `rcl_param_descriptor_t`  array.

Reply:
Fair point

Reply:
This comment still applies i.e. no need to have the parameter name in three (3) places.

Reply:
I removed the name field in `rcl_param_descriptor_t` instead since it made more sense following the parallel logic of the code with the parameter values.